pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -828,18 +832,18 @@ func (plugin *kubenetNetworkPlugin) disableContainerDAD(id kubecontainer.Contain
 // given a n cidrs assigned to nodes,
 // create bridge configuration that conforms to them
 func (plugin *kubenetNetworkPlugin) getRangesConfig() string {
-	createRange := func(thisNet *net.IPNet) string {
+	createRange := func(thisNet, thisGw string) string {
 		template := `
 [{
 "subnet": "%s",
 "gateway": "%s"
 }]`
-		return fmt.Sprintf(template, thisNet.String(), thisNet.IP.String())
+		return fmt.Sprintf(template, thisNet, thisGw)
 	}
 
 	ranges := make([]string, len(plugin.podCIDRs))
 	for idx, thisCIDR := range plugin.podCIDRs {
-		ranges[idx] = createRange(thisCIDR)
+		ranges[idx] = createRange(thisCIDR.String(), plugin.podGateways[idx].String())
```

Comment:
might be nicer to say `plugin.podCIDRs[idx]`, for more-obviously parallelism

Reply:
podCIDR and podGateway have a corresponding relationship,  will it be better to combine  them in a new struct?